### DISCUSSION TITLE
FWI with density and/or Q-factor

### DISCUSSION THREAD

**Discussion by kerim371** [slimgroup/JUDI.jl#153] [General]
Hi,

I'm wondering is it possible to run FWI with density (and possibly Q-factor) so that after each iteration both slowness and density is updated?
I can see there is an [fwi_example_constraints.jl](https://github.com/slimgroup/JUDI.jl/blob/f1ea0232feaa0c157ce9272f9c4760bca0f06d0d/examples/scripts/fwi_example_constraints.jl#L38-L56) - is this what I'm looking for?

**Comment by mloubout** [comment]
You can currently propagate with density and Q but only the sensitivity w.r.t velocity are implemented. For density you can usually get away with Gardner relationship
